File: .github/atoma/tools/scripts/get_config_value.py
#!/usr/bin/env python3
"""Print a dotted-path value from .github/atoma/config.json.

Usage:
  get_config_value.py <dotted.path> [default]

Examples:
  get_config_value.py agents.engineer.max_iterations 30
  get_config_value.py labels.in_progress atoma/in-progress
"""

from __future__ import annotations

import sys

from atoma_config import load_config


def main() -> None:
    if len(sys.argv) < 2:
        print("usage: get_config_value.py <dotted.path> [default]", file=sys.stderr)
        sys.exit(2)

    path = sys.argv[1]
    default = sys.argv[2] if len(sys.argv) > 2 else ""

    node = load_config()
    for key in path.split("."):
        if isinstance(node, dict) and key in node:
            node = node[key]
        else:
            print(default)
            return
    print(node)


if __name__ == "__main__":
    main()

File: src/atoma/tools/scripts/get_config_value.ts
#!/usr/bin/env bun
/**
 * Print a dotted-path value from .github/atoma/config.json.
 *
 * Usage:
 *   bun run get_config_value.ts <dotted.path> [default]
 *
 * Examples:
 *   bun run get_config_value.ts agents.engineer.max_iterations 30
 *   bun run get_config_value.ts labels.in_progress atoma/in-progress
 *
 * NOTE: this is Atoma's own copy (called by launch_sub_agent.ts, a
 * same-directory sibling). An independent, functionally-identical copy also
 * lives at `.github/scripts/get_config_value.ts` (source-tracked in
 * src/scripts/, called by atoma-runner.wac.ts) -- deliberately duplicated
 * rather than shared so each of `.github/atoma/**` and `.github/scripts/**`
 * stays a fully self-contained tree with no cross-references between them.
 */
import { loadConfig } from "./lib/config.ts";

function main(): void {
  const [path, fallback = ""] = Bun.argv.slice(2);
  if (!path) {
    console.error("usage: get_config_value.ts <dotted.path> [default]");
    process.exit(2);
  }

  let node: unknown = loadConfig();
  for (const key of path.split(".")) {
    if (node && typeof node === "object" && key in node) {
      node = (node as Record<string, unknown>)[key];
    } else {
      console.log(fallback);
      return;
    }
  }
  console.log(node);
}

if (import.meta.main) main();
